core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -0,0 +1,233 @@
+          allPlayerStats.get(player.getBukkit().getName()).killstreakMax);
+      allDeaths.put(
+          player.getBukkit().getDisplayName(),
+          allPlayerStats.get(player.getBukkit().getName()).deaths);
+      allBowshots.put(
+          player.getBukkit().getDisplayName(),
+          allPlayerStats.get(player.getBukkit().getName()).longestBowKill);
+    }
+    Component killMessage = getKillsMessage(sortStats(allKills));
+    Component killstreakMessage = getKillstreakMessage(sortStats(allKillstreaks));
+    Component deathMessage = getDeathsMessage(sortStats(allDeaths));
+    Map.Entry<String, Integer> bestBowshot = sortStats(allBowshots);
+    Component bowshotMessage = getBowshotMessage(bestBowshot);
+
+    for (MatchPlayer viewer : match.getPlayers()) {
+      viewer.sendMessage(
+          Components.fromLegacyText(
+              ComponentUtils.horizontalLineHeading(
+                  ChatColor.YELLOW + "Best stats this match",
+                  ChatColor.WHITE,
+                  ComponentUtils.MAX_CHAT_WIDTH)));
+
+      viewer.sendMessage(killMessage);
+      viewer.sendMessage(killstreakMessage);
+      viewer.sendMessage(deathMessage);
+      if (bestBowshot.getValue()
+          != 0) { // Prevent this from showing if bows are not used on the map played
+        viewer.sendMessage(bowshotMessage);
+      }
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map.Entry<String, Integer> sortStats(HashMap<String, Integer> map) {
+    Object[] a = map.entrySet().toArray();
+    Arrays.sort(
+        a,
+        (Comparator)
+            (o1, o2) ->
+                ((Map.Entry<String, Integer>) o2)
+                    .getValue()
+                    .compareTo(((Map.Entry<String, Integer>) o1).getValue()));
+    return (Map.Entry<String, Integer>) a[0];
+  }
+
+  Component getKillsMessage(Map.Entry<String, Integer> topResult) {
+    return new Component(
+        new PersonalizedTranslatable(
+                "stats.kills",
+                new PersonalizedText(topResult.getKey()),
+                new PersonalizedText(Integer.toString(topResult.getValue()), ChatColor.GREEN))
+            .render());
+  }
+
+  Component getKillstreakMessage(Map.Entry<String, Integer> topResult) {
+    return new Component(
+        new PersonalizedTranslatable(
+                "stats.killstreak",
+                new PersonalizedText(topResult.getKey()),
+                new PersonalizedText(Integer.toString(topResult.getValue()), ChatColor.GREEN))
+            .render());
+  }
+
+  Component getDeathsMessage(Map.Entry<String, Integer> topResult) {
+    return new Component(
+        new PersonalizedTranslatable(
+                "stats.deaths",
+                new PersonalizedText(topResult.getKey()),
+                new PersonalizedText(Integer.toString(topResult.getValue()), ChatColor.RED))
+            .render());
+  }
+
+  Component getBowshotMessage(Map.Entry<String, Integer> topResult) {
```

Comment:
I think we should make a class representing `Map.Entry<String, Integer> topResult`.

Reply:
A `Record` here would be amazing, too bad we're on java 8 and not 14 :cry:

Reply:
What do you mean Molka? Represent `topResult` as class in which way?

Reply:
Make a class with two fields, `String` and `int`. A dedicated class with named fields is easier to understand than `Map.Entry`.

Reply:
This should be resolved in coming update, thanks :)